components/automate-cli/cmd/chef-automate/generate-a2harb.go

```diff
@@ -4,18 +4,31 @@ package main
 
 import (
 	"bytes"
+	"container/list"
 	"html/template"
 	"io/ioutil"
+	"net"
 	"strings"
 
 	"github.com/chef/automate/components/automate-cli/pkg/status"
 	ptoml "github.com/pelletier/go-toml"
 	"github.com/pkg/errors"
 )
 
-func readConfigAndWriteToFile() error {
-	writer.Printf("reading configs from toml file")
+func getSingleErrorFromList(l *list.List) error {
+	errorString := ""
+	for e := l.Front(); e != nil; e = e.Next() {
+		errorString = errorString + "\n" + e.Value.(string)
+	}
+	return errors.New(errorString)
+}
+
+func readConfigAndWriteToFile(configPath string) error {
```

Comment:
This function also needs to decide on different architectures. We should use deployer and add the respective functions in the struct.